src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountIf.cs

```diff
@@ -64,10 +64,21 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
             var fValid = CheckInvocation(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
             Contracts.Assert(returnType == DType.Number);
 
-            // Ensure that all the args starting at index 1 are booleans.
+            // Ensure that all the args starting at index 1 are booleans or coerece to boolean if they are OptionSetValues.
             for (var i = 1; i < args.Length; i++)
             {
-                if (!DType.Boolean.Accepts(argTypes[i]))
+                if (argTypes[i] == DType.OptionSetValue && CheckType(args[i], argTypes[i], DType.Boolean, DefaultErrorContainer, out var matchedWithCoercion))
```

Comment:
Why go through CheckType? We could just use `argTypes[i].CoercesTo(Dtype.Boolean)` at this point, I think.